connection.go

```diff
@@ -510,13 +510,13 @@ func (c *Connection) SendSystemError(id uint32, span Span, err error) error {
 		tracing: span,
 		message: GetSystemErrorMessage(err),
 	}); err != nil {
-
 		// This shouldn't happen - it means writing the errorMessage is broken.
 		c.log.WithFields(
 			LogField{"remotePeer", c.remotePeerInfo},
 			LogField{"id", id},
 			ErrField(err),
 		).Warn("Couldn't create outbound frame.")
+		c.opts.FramePool.Release(frame)
```

Comment:
for each place we call `Release`, having a comment with context for why we release would be useful.
E.g.,
`// Frame allocated is never written to `sendCh`, so we manually release it to the pool.`

Reply:
added comments above the allocation since we now use defer to release, with the code directly under the alloc